pkg/apis/azureprovider/v1beta1/azuremachineproviderconfig_types.go

```diff
@@ -96,6 +96,13 @@ type AzureMachineProviderSpec struct {
 	// SecurityProfile specifies the Security profile settings for a virtual machine.
 	// +optional
 	SecurityProfile *SecurityProfile `json:"securityProfile,omitempty"`
+
+	// AcceleratedNetworking enables or disables Azure accelerated networking. If omitted, it will be set based on
+	// whether the requested VMSize supports accelerated networking.
```

Comment:
If omitted, it defaults to off right? This comment doesn't align with what I expected

Also, since you are wanting this to default to false, there's no need to differentiate between a null value and a false value, you can just make this a `bool` not `*bool` no?

Reply:
I copied the variable along with the comment from the upstream repository here. I will update the code comment.
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/api/v1alpha4/azuremachine_types.go#L97-L102

I think defaulting to `false` keeps compatibility with existing clusters. I think the behavior upstream is to turn the feature on if the VM supports it. This will be something to take into account if/when moving to the upstream provider.

Reply:
I will attempt to switch to bool and retest.

Reply:
Tested the latest code and verified the accelerated networking was enabled.

Reply:
Thanks, I think it makes sense to keep the setting as is for existing users, we can revisit this later during the migration process though